# <div style="width:100%;text-align: center;color:darkblue;font-size:45px">  Global Comoodity Trade </div>
 
    
*Author: Yousef Moterassed*  
*September 2022*
  
***

# Introduction

These days it is inevitable to be involved with the economy and politics since it strongly affects our day-to-day life. So it might be helpful to have a realistic and data-driven sight of the actual situation of the global economy and trading exchanges.

My name is [Yousef Moterassed](https://www.kaggle.com/yousefmoterassed), a Persian borned boy who lives in Samba-Land 🇧🇷. I am working in the trade and business field for many years and after stepping into the Data Science and Data Analysis road, decided to combine these knowledges to be able to uncover some hidden information behind the curtain of the data, and release some data-driven information publicly in which it might help myself and our society to increase their understanding and common knowledge about what's going on in our world's economy.

I hope this project which has been done with Python in the Kaggle platform, would help some of you to have a better vision of what's going on in our world's economy and trades.

## Who are the audience of this work?

This project has been done in hope to help to improve the **public knowledge** in the first place. Although some statistical or mathematical terms and calculations in some contents might seem a bit difficult or boring to understand, I will try to facilitate them with simplified visualizations and will be explained in plain English within the <a href='#1'>Explanation and Observation</a> section.

However this is my hope that through the ease of using this project some of our statisticians, civil and public servants, politicians and economists, to help the progress of our societies with their deeper and more technical works.

# About the data

This dataset was kindly published by the **United Nations** Statistics Division on the **[UNData](http://data.un.org/Explorer.aspx)** site. You can find the original dataset [here](http://data.un.org/Explorer.aspx).

This dataset contain about **8,225,870 entries** and around **1.23 GB** of data.
Some of the numbers on this dataset are more trustworthy than others. I'd expect that British tea imports are fairly accurate, but doubt that Afghanistan exported exactly 51 sheep in 2016. Can we identify which nations appear to have the most trustworthy data? Which industries? 

I will try to reach out to some useful and reliable conclusions at the end and surely will keep updating this notebook with fresh ideas and new techniques I would learn during my DS journey, and I hope you would not withhold your conclusions and opinions about this analysis.


## Columns explanation

This dataset contains 10 major columns and is defined as below:
1. **country_or_area**: Name of the country or the area in which we will rename it to "Country" for our analysis
2. **year**: This is an annual report, so it is the year in which the trade has taken place
3. **comm_code**: Per the World Customs Organization: The Harmonized Commodity Description and Coding System generally referred to as "Harmonized System" or simply "HS" is a multipurpose international product nomenclature developed by the **World Customs Organization** (WCO). It comprises about 5,000 commodity groups; each identified by a six digit code, arranged in a legal and logical structure and is supported by well-defined rules to achieve uniform classification. For more, see [here](http://www.wcoomd.org/en/topics/nomenclature/overview/what-is-the-harmonized-system.aspx)
4. **commodity**: The description of a particular commodity code, i.e. "Horses, live pure-bred breeding"
5. **flow**: Flow of trade i.e. Export, Import, Re-Import
6. **trade_usd**: Value of the trade in US Dollars ($)
7. **weight_kg**: Weight of the commodity in is the SI unit of mass, Kilograms (Kg)
8. **quantity_name**: A description of the quantity measurement type given the type of item (i.e. Number of Items, Weight in Kilograms, etc.)
9. **quantity**: Count of the quantity of a given item based on the Quantity Name
10. **category**: Category to identify commodity i.e "02 Meat and edible meat offal"


## Licence
License
[Per the UNData terms of use](http://data.un.org/Host.aspx?Content=UNdataUse): all data and metadata provided on UNdata’s website are available free of charge and may be copied freely, duplicated and further distributed provided that [UNdata](http://data.un.org/Explorer.aspx) is cited as the reference.


# What is the purpose of my project?

I have learned from the most impressive person in my life whom her legacy and lessons are in my ears everyday and most imortant of them is that we came to this world to create, not just to consume. And we have to know somethings before creating anything for other human being to use or consume.

Why are we doing this?
For who we are doing it?
When is better to do it?

### Questions to be answered:
1. when was the year with the most import and export records?
2. Visualizing each year import and export records(Out of the item above)
3. Global trades are mostly in import or export segment?
4. Which countries have the most participation in import and export?
5. Verify them in each segment/import and export
6. Which items traded in the most quantity or weight?
7. Put two or three countries in three groups of high/mid/low trade segment and analyze their trends.
8. what was the impact of the covid on the market?
9. Highest exporters imported what products in return?
10. Take a look at your mothersland, iran and brazil segments and trends. Maybe in another notebook 
11. What is the value of export and import traded per year?
12. Does this value affect the world deveoloping velocity?
13. which category is in the high demand?
14. what are the top categories for each countries in those groups you defined?
15. What is the value of the trades per year for those countries?
16. Does those values helped the countries to develop?
17. * Which countries exported more than they imported?
18. Whiat products or services serves the world better?
19. Is it real that feeding human being is the priority for the countries? What are ther references?

# Preparing our dataset

Here is the very first step of our work, we simply import libraries and call our dataset in a dataframe called `df`

In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import os
print(os.listdir("../input"))
df= pd.read_csv('../input/global-commodity-trade-statistics/commodity_trade_statistics_data.csv',low_memory=False)

['global-commodity-trade-statistics']


Let us have a quick look at this large data set with only first five rows.

In [2]:
df.head()

country_or_area  year comm_code  \
0     Afghanistan  2016    010410   
1     Afghanistan  2016    010420   
2     Afghanistan  2008    010210   
3         Albania  2016    010290   
4         Albania  2016    010392   

                                         commodity    flow  trade_usd  \
0                                      Sheep, live  Export       6088   
1                                      Goats, live  Export       3958   
2          Bovine animals, live pure-bred breeding  Import    1026804   
3  Bovine animals, live, except pure-bred breeding  Import    2414533   
4    Swine, live except pure-bred breeding > 50 kg  Import   14265937   

   weight_kg    quantity_name  quantity         category  
0     2339.0  Number of items      51.0  01_live_animals  
1      984.0  Number of items      53.0  01_live_animals  
2      272.0  Number of items    3769.0  01_live_animals  
3  1114023.0  Number of items    6853.0  01_live_animals  
4  9484953.0  Number of items   96040.0  01_live_animals

First I would like to **rename** the column name of the **"country_or_area" to "country"** in which would facilitate our works.

In [3]:
df.rename(columns = {'country_or_area':'country'}, inplace = True)

# Descriptive and Exploratory Statistics (EDA)

This is important to understand that these days the statistics and mathematical calculations behind it is the inseparable part of any serious analysis. Luckily the efforts of prior scientists made our job way easier than before, since there are some functions in Python like `describe()` and `info()` to calculate all the important statistical tools for us, also to see the data types and entries.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8225871 entries, 0 to 8225870
Data columns (total 10 columns):
 #   Column         Dtype  
---  ------         -----  
 0   country        object 
 1   year           int64  
 2   comm_code      object 
 3   commodity      object 
 4   flow           object 
 5   trade_usd      int64  
 6   weight_kg      float64
 7   quantity_name  object 
 8   quantity       float64
 9   category       object 
dtypes: float64(2), int64(2), object(6)
memory usage: 627.6+ MB


We can see that the column year is parsed as integer, so we will convert it to datetime format and then verify the description of this dataset

In [5]:
df['year']= pd.to_datetime(df['year'], format='%Y')

In [6]:
df.describe(include='all', datetime_is_numeric=True)

country                           year comm_code        commodity  \
count     8225871                        8225871   8225871          8225871   
unique        209                            NaN      5040             5031   
top     Australia                            NaN     TOTAL  ALL COMMODITIES   
freq       307627                            NaN      9185             9185   
mean          NaN  2004-10-07 17:50:05.544289280       NaN              NaN   
min           NaN            1988-01-01 00:00:00       NaN              NaN   
25%           NaN            1999-01-01 00:00:00       NaN              NaN   
50%           NaN            2005-01-01 00:00:00       NaN              NaN   
75%           NaN            2011-01-01 00:00:00       NaN              NaN   
max           NaN            2016-01-01 00:00:00       NaN              NaN   
std           NaN                            NaN       NaN              NaN   

           flow     trade_usd     weight_kg        quantity_name  \
count   8225871  8.225871e+06  8.097396e+06              8225871   
unique        4           NaN           NaN                   12   
top      Import           NaN           NaN  Weight in kilograms   
freq    4848524           NaN           NaN              6617980   
mean        NaN  9.505370e+07  3.556503e+07                  NaN   
min         NaN  1.000000e+00  0.000000e+00                  NaN   
25%         NaN  1.107200e+04  1.089000e+03                  NaN   
50%         NaN  1.534160e+05  2.689800e+04                  NaN   
75%         NaN  1.895476e+06  4.474432e+05                  NaN   
max         NaN  2.443311e+12  1.860133e+12                  NaN   
std         NaN  7.074154e+09  2.479277e+09                  NaN   

            quantity                         category  
count   7.921014e+06                          8225871  
unique           NaN                               98  
top              NaN  95_toys_games_sports_requisites  
freq             NaN                           100000  
mean    2.671470e+08                              NaN  
min     0.000000e+00                              NaN  
25%     1.062000e+03                              NaN  
50%     2.712000e+04                              NaN  
75%     4.623620e+05                              NaN  
max     1.026357e+15                              NaN  
std     3.739231e+11                              NaN

This function gives us very interesting information and description about this dataset, such as: 
* the number of countries recorded in this dataset which is counted as **209 distinct countries**. 
* Also, we can see the interval of these records which is between **1988 to 2016**.
* The top country in all segments appears to be **Australia** 
* The import records seem to be more applied than exports.
* There are 98 product categories and the **category code 95** *Toys, Games, Sports requisites* is in the highest records of deals.

We will dive deeper on other items later in other sections.

<a id="1"></a><h1>Explanation and Observation</h1>

[1].   **Variance**: the difference between the observation average and sample average.